### PR TITLE
feat: pause FT transfers

### PR DESCRIPTION
**Pull request by karim-en**
## Description
Add the possibility of pausing the ft transfers for the internal eth-connector; this is needed to avoid pausing the whole Aurora engine during migration.
A new flag, `PAUSE_FT`, has been added that pauses all the public methods that could change the internal balances of the eth-connector token [`ft_transfer`, `ft_transfer_call`, `storage_deposit`, `storage_unregister`].

## Testing
Some tests have been added to test the pause and unpause logic.

### CODE REVIEW

File: engine-tests/src/utils/workspace.rs [hunk 1]
```diff
@@ -236,6 +238,40 @@ pub async fn transfer_nep_141(
     Ok(())
 }
 
+pub async fn transfer_call_nep_141(
```
Comment (aleksuss) [inline engine-tests/src/utils/workspace.rs:241]: I believe it should be annotated with `#[cfg(not(feature = "ext-connector"))]`

File: engine/src/contract_methods/connector/errors.rs [hunk 2]
```diff
@@ -118,6 +119,7 @@ impl AsRef<[u8]> for FtTransferCallError {
             Self::InsufficientAmountForFee => errors::ERR_NOT_ENOUGH_BALANCE_FOR_FEE,
             Self::Transfer(e) => e.as_ref(),
             Self::BalanceOverflow(e) => e.as_ref(),
+            Self::Paused => errors::ERR_PAUSED,
```
Comment (aleksuss) [inline engine/src/contract_methods/connector/errors.rs:122]: I would introduce a new error: `error::ERR_FT_PAUSED`.

**Review by aleksuss — COMMENTED** [review]
Overall looks good with couple nitpicks.

File: engine-tests/src/utils/workspace.rs (hunk 1 above)
Comment (karim-en) [reply engine-tests/src/utils/workspace.rs:259]: 219007bf9b048c152dd5f94abafd7126e9316331

File: engine/src/contract_methods/connector/errors.rs (hunk 2 above)
Comment (karim-en) [reply engine/src/contract_methods/connector/errors.rs]: 4a6accd250b4312bc51315f4b452c61170f4c9d0

File: engine-tests/src/utils/workspace.rs [hunk 3]
```diff
@@ -236,6 +238,40 @@ pub async fn transfer_nep_141(
...
+    nep_141: &AccountId,
+    source: &Account,
+    dest: &str,
+    amount: u128,
+    msg: &str,
+    storage_deposit: bool,
```
[the remark this answers is not in the document]
Comment (karim-en) [reply engine-tests/src/utils/workspace.rs]: The storage deposit was moved to a separate function d90c9c4d33970bc17d9e28b3acb5154a9316cb76

**Review by birchmd — APPROVED** [review]

**Review by aleksuss — APPROVED** [review]

**Review by mrLSD — APPROVED** [review]

**Pull request merged**